modules/management-group-bootstrap/readers.tf

```diff
@@ -15,10 +15,17 @@ resource "azurerm_role_assignment" "readers" {
   role_definition_name = "Reader"
 }
 
-#resource "azurerm_role_assignment" "app_gateway_backend_health_reader" {
-#for_each = var.groups
+locals {
+  custom_role_assignments = [
+    "Application Gateway Backend Health Reader"
+  ]
+}
+
 
-#principal_id         = azuread_group.readers[each.value.id].object_id
-#scope                = "/providers/Microsoft.Management/managementGroups/${each.value.id}"
-#role_definition_id   = "159469f4-6e46-4b09-aa73-8e2f243aa784"
-#}
+resource "azurerm_role_assignment" "custom_role_assignments_readers" {
+  for_each = {for group_key, group_value in var.groups : { for role in local.custom_role_assignments : group_key => role }}
+
+  principal_id       = azuread_group.readers[each.key].object_id
```

Comment:
why use a different format to subscriptions? i.e. that has the principal id as part of the object:

```hcl
  custom_role_assignments = {
    "Application Gateway Backend Health Reader" = {
      principal_id = azuread_group.groups["Reader"].id
      scope        = "/subscriptions/${azurerm_subscription.this.subscription_id}"
    },
  }
```

(scope isn't needed here and should probably be optional in the subscriptions one)

Reply:
I wanted similar behaviour to the Subscriptions Module (just adding to a list for each assignment), not sure we'll ever need to, but this also allows different roles to be assigned to the Readers and Owners groups (haven't added any tf code for owners because i don think it's needed). 

I've been staring at this for too long, but I'm sure the difference in the `custom_role_assignments` lists is needed as the subscription module builds out a single subscription, whereas the MGMT/Enterprise module is takes a list of MGMT groups so I need to get the list of groups and then the list of assignments.